src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java

```diff
@@ -204,23 +208,45 @@ private void saveAndInvokeChsKafkaApi(String contextId, String companyNumber,
 
     /**
      * Delete charge from company mortgages.
+     * @param contextId the x-request-id.
      * @param chargeId the charge identifier.
+     * @param companyNumber the company number.
      */
-    public void deleteCharge(String contextId, String chargeId) throws Exception {
+    public void deleteCharge(String companyNumber,
+                             String contextId,
+                             String chargeId) {
         try {
-            if (!chargesRepository.existsById(chargeId)) {
+            Optional<ChargesDocument> chargesDocumentOptional =
+                    chargesRepository.findById(chargeId);
+
+            if (chargesDocumentOptional.isEmpty()) {
                 throw new IllegalArgumentException(String.format(
```

Comment:
If charges document does not exists, it should be 400 bad request response back, not sure if this exception results in 400?

Reply:
I will check the type of exception and set the appropriate code in controller. I think all the exceptions stuff will be streamlined part of a new story which is yet to be worked DSND-781. All status code related exceptions will be covered in this story.

Reply:
sure, lets do in 781